modules/imgproc/src/color.cpp

```diff
@@ -5917,9 +5893,12 @@ static inline softfloat applyInvGamma(softfloat x)
                 pow(xd, softdouble::one()/gammaPower)*(softdouble::one()+gammaXshift) - gammaXshift);
 }
 
+static Mutex tablesInitMtx;
+
 static void initLabTabs()
 {
     static bool initialized = false;
+    AutoLock tablesInitLock(tablesInitMtx);
     if(!initialized)
```

Comment:
Please apply this pattern:
https://en.wikipedia.org/wiki/Double-checked_locking